### PR TITLE
22806 update document management page

### PR DESCRIPTION
**Pull request by flutistar**
# 22806

Updated document index table.

### CODE REVIEW

File: document-service/documents-ui/src/components/HelpToggleContainer.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<script setup lang="ts">
+  const isHelpContentOpen = ref(false)
+  const hasBottomHideToggle = ref(true)
```
Comment (cameron-eyds) [inline document-service/documents-ui/src/components/HelpToggleContainer.vue:3]: Future-state prop potentially?

File: document-service/documents-ui/src/components/documentsTable/index.vue [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 }
+
+
+const searchDateRange = ref()
+const searchDocumentClass = ref()
+const documentRecordColumns = ref([])
+
+onMounted(() => {
+  documentRecordColumns.value = documentResultColumns()
+})
```
Comment (cameron-eyds) [inline document-service/documents-ui/src/components/documentsTable/index.vue:21]: We can probably use documentResultColumns() directly without the hook but we can refine this, as we will tie in with the request for filtered results (or unfiltered). No action required, just a note for future.

File: document-service/documents-ui/src/components/documentsTable/index.vue [hunk 3]
```diff
@@ -20,28 +29,142 @@ const openDocumentRecord = (searchResult: DocumentInfoIF) => {
...
+        </div>
+        </template>
+        <template #actions-header = "{column}">
+          <div class="px-2">
+            {{ column.label }}
+          </div>
+          <UDivider class="my-3"/>
+          <div>
+          <div class="h-8">
+          </div>
```
Comment (cameron-eyds) [inline document-service/documents-ui/src/components/documentsTable/index.vue:165]: Is this for spacing? If so you can also `<div class="h-8" />` or use a NuxtUi native component for spacing. either or.

File: document-service/documents-ui/src/components/documentsTable/index.vue [hunk 4]
```diff
@@ -58,14 +181,28 @@ const openDocumentRecord = (searchResult: DocumentInfoIF) => {
...
+               class="flex align-center"
                @click="downloadFileFromUrl(row.documentUrls[i], file)"
              >
+              <UIcon name="i-mdi-file-download" class="w-5 h-5" />
               {{ file }}
             </ULink>
           </span>
+          <span
+            v-if="row.consumerFilenames.length > 1"
+          >
```
Comment (cameron-eyds) [inline document-service/documents-ui/src/components/documentsTable/index.vue:197]: Very minor, but general rule of thumb:
if only 1 prop on an element, can single line it
if multiple: multi lines

no biggie though, just for reference

**Review by cameron-eyds — COMMENTED** [review]
Exceptional work sir!

Few minor items to address here:
 Linting/Unit testing 
 
 Few future items to make a note of:
 Tie in Date Range pre-select functionality (looks like they are missing values, or does it work as is?)
 Help content to be provided from design. 
 Refactor that initial population of the table (will come with filter request task)

**Review by cameron-eyds — COMMENTED** [review]
Also add to future todos: Search Filter models. 
Essentially the values bound to our filter fields which we will use in our request, we can do this after though!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://documents-ui-dev--pr-49-y1c45o4h.web.app

**Review by cameron-eyds — COMMENTED** [review]
Code is looking good, just a few small things:

Screenshot 2024-09-18 at 6 30 15 AM

Tooltips are not centered off the icon

Screenshot 2024-09-18 at 6 30 21 AM

Document Type selector is giving the options for the classes not the types. These options would be used for this selector in the header. 

Screenshot 2024-09-18 at 6 33 10 AM

**Review by cameron-eyds — APPROVED** [review]
Looks good, just fix those small lint items before squash/merging!

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://documents-ui-dev--pr-49-y1c45o4h.web.app

**Review by doug-lovett — APPROVED** [review]
Looks good.

**Pull request merged**